tests/std/tests/P0896R4_views_join/test.cpp

```diff
@@ -745,13 +745,17 @@ int main() {
 
     { // P2328 range of prvalue string using global function
         assert(ranges::equal(views::iota(0u, 5u) | views::transform(ToString) | views::join, expected));
+#if !(defined(_DEBUG) && defined(__EDG__)) // TRANSITION, not yet reported, see also GH-1566

```

Comment:
I'd like to report a bug to track here before merging; do you have a minimal repro for this issue?

Reply:
I haven't managed to remove `string` dependency. Here's what I get so far:

```c++
#include <memory> // for construct_at
#include <string>
using namespace std;

struct fake_non_propagating_cache {
    bool engaged = false;
    union { string val; };

    constexpr fake_non_propagating_cache() {}

    constexpr void emplace() {
        construct_at(&val, "a"s);
        engaged = true;
    }

    constexpr string& operator*() { return val; }

    constexpr ~fake_non_propagating_cache() {
        if (engaged) {
            val.~string();
        }
    }
};

constexpr bool test() {
    fake_non_propagating_cache str;
    str.emplace();

    auto it = (*str).begin();
    (void)*it;

    return true;
}

int main() {
    static_assert(test());
}
```

Reply:
Slightly reduced example:
```C++
#include <string>

constexpr bool test() {
    union string_holder {
        constexpr string_holder() : s("SSO") {} // fine with long strings
        constexpr ~string_holder() {
            s.~basic_string();
        }

        std::string s;
    };
    string_holder sh;
    (void) *sh.s.begin();
    return true;
}

int main() {
    static_assert(test());
}
```
Seemingly related to lifetime of union members.

Reply:
"Reduced" to:

```c++
namespace std {
    struct string;

    struct _Container_proxy { const string* _Mycont; };

    struct iterator {
        _Container_proxy* _Myproxy;
        const char* _Ptr;

        constexpr void operator*() const;
    };

    union _Bxty { char _Buf[16]{}; };

    struct string {
        constexpr string() : _Myproxy{new _Container_proxy{this}} {}

        string(const string&) = delete;
        string& operator=(const string&) = delete;

        constexpr ~string() { delete _Myproxy; }

        constexpr iterator cbegin() const {
            return iterator{_Myproxy, _Bx._Buf};
        }

        _Container_proxy* _Myproxy;
        _Bxty _Bx;
    };

    constexpr void iterator::operator*() const {
        auto _Contptr = _Myproxy->_Mycont->_Bx._Buf;
        (void)(_Ptr < _Contptr + 4);
    }
} // namespace std

constexpr bool test() {
    union string_holder {
        constexpr ~string_holder() { s.~string(); }
        std::string s{};
    };
    string_holder sh;
    *sh.s.cbegin();
    return true;
}

static_assert(test());
```

and filed as VSO-1948896 "Catastrophic Edge error involving constexpr std::string in a union". (I'll update the comment in the source.)